### DISCUSSION TITLE
建议DDE增加支持vnc局域网的“桌面共享”功能

### DISCUSSION THREAD

**Discussion by yanchenyuvip** [linuxdeepin/developer-center#5620] [Development Questions | 开发答疑] [answered]
流行的 Linux桌面环境Gnome，KDE，XFCE等都在系统设置中都有预装“桌面共享”功能即支持vnc协议的服务端，用户开箱即用，只需勾选打开，再其它平台用vnc viewer输入ip连接即可，这已经成为一种用户习惯。我看到deepin和uos维护的推荐文档中建议用户使用x11vnc，但实际上这个x11vnc这个项目已经可以称得上年久失修https://github.com/LibVNC/x11vnc
“x11vnc”已经连续4年未发布新版本，同时该项目也在寻找维护者，总体来说它的可靠程度已经下将，后期它将可能无法满足不断迭代的deepin。

备注：uos自带的“远程协助”试过Window vnc viewer局域网是无法连接的，我建议对“远程协助”做扩展vnc协议支持，方便局域网用户直接连接，而不是一大堆教程和步骤让小白再去尝试。

**Comment by BLumia** [comment, marked as answer]
KDE 的 Krfb 在 DDE (x11)下其实是可以用的，另外 Rust Desk 支持局域网内的屏幕共享（以及文件传输等功能）。在 DDE 打算提供类似功能之前，这些可以作为替代。Krfb 之类的基于 vnc 的方案不确定是不是到 wayland 就没法用了 = =

cc @meiyixiang

**Reply by yanchenyuvip** [reply]
感谢

**Reply by yanchenyuvip** [reply]
realvnc已经明确回复暂时wayland 暂时无法支持vnc，建议deepin采用Ubuntu目前采用的方案要么登录界面切Xorg要么支持微软的msrd协议，总之尽量要为用户官方提供一个稳定且开箱即用的远程桌面共享功能。
![image](https://github.com/linuxdeepin/developer-center/assets/44298325/80b7ca3f-86b6-456c-860c-0dc6349c98da)

[https://help.realvnc.com/hc/en-us/articles/4417193011857-How-do-I-disable-Wayland-to-use-VNC-Connect-](url)
